app/post.go

```diff
@@ -726,15 +726,23 @@ func (a *App) publishWebsocketEventForPermalinkPost(post *model.Post, message *m
 		return false, err
 	}
 
+	permalinkPreviewedChannel, err := a.GetChannel(previewedPost.ChannelId)
+	if err != nil {
+		if err.StatusCode == http.StatusNotFound {
+			mlog.Warn("channel containing permalinked post not found", mlog.String("referenced_channel_id", previewedPost.ChannelId))
+			return false, nil
+		}
+		return false, err
+	}
+
+	permalinkPreviewedPost := post.GetPreviewPost()
```

Comment:
Wondering how this could impact performance with the extra GetChannel and GetPreviewPost calls? This ticket already had some concerns https://mattermost.atlassian.net/browse/MM-42810 around this function

Reply:
@BenCookie95 This actually reduces the number of `GetPreviewPost` and `GetChannel` calls by doing them here instead of in `SanitizePostMetadataForUser` in `post_metadata.go`. Before, we were making these calls for each user in a channel before publishing the message, but now these calls are made once here and their values are passed to the new function `sanitizePostMetadataForUserAndChannel`. I think this should help with some of the performance issues in [MM-42810](https://mattermost.atlassian.net/browse/MM-42810).